fl4health/clients/ditto_mkmmd_client.py

```diff
@@ -0,0 +1,483 @@
+from logging import INFO
+from pathlib import Path
+from typing import Dict, List, Optional, Sequence, Tuple
+
+import torch
+import torch.nn as nn
+from flwr.common.logger import log
+from flwr.common.typing import Config, NDArrays, Scalar
+from torch.optim import Optimizer
+
+from fl4health.checkpointing.checkpointer import TorchCheckpointer
+from fl4health.clients.basic_client import BasicClient
+from fl4health.losses.mkmmd_loss import MkMmdLoss
+from fl4health.model_bases.moon_base import MoonModel
+from fl4health.parameter_exchange.full_exchanger import FullParameterExchanger
+from fl4health.utils.losses import EvaluationLosses, LossMeterType, TrainingLosses
+from fl4health.utils.metrics import Metric
+
+
+class DittoClient(BasicClient):
```

Comment:
I like the idea of having a separate file/class for this implementation and I think we should do something sort of similar for the FENDA and MOON implementations. That way we keep the original implementations simpler and have a more complex version. 

However, I have two suggestions here. The first is to name the class something different (maybe just `DittoMkMmdClient`), so that there isn't confusion when instantiating a class. Second, I think we should inherit from `DittoClient` here (rather than just `BasicClient`), as there are a lot of methods that can be reused and the remainder can be override.

Reply:
I like the idea so much and I think in fact we should do for all Moon, Fenda and ditto methods. I can do refactor now as it shouldn't take so much time.